### PR TITLE
add ImmediateContainer, Quantity and SimplyUnit, clarify Unit use

### LINKED ISSUE #1343: Clarify unit description
Current:

> A description of the unit in which the supplies, services or works are provided (e.g. hours, kilograms) and the unit-price

From the research for the medicine extension, ISO-11240 (Health informatics — Identification of medicinal products — Data elements and structures for the unique identification and exchange of units of measurement) notes:

> Units of measurement shall be distinguished from other notions of the word “unit”, such as counted entities or “units of presentation”.

> NOTE 1: Units of presentation are conventional terms that can be used to describe the properties of a pharmaceutical product, e.g. tablet, vial, puff. They are not the same as units of measurement. Units of measurement are always defined through some reference quantities, which units of presentation are not.

> NOTE 2: Concepts like “tablet”, “patch”, “bottle”, etc., are often used to characterize the various types of countable items. In the context of measurements, they simply represent the quantity “1”, corresponding to the number of items they represent.

In the current description, we only give units of measurement as examples. We also use "unit(s) of measure" in the description of the unit's `scheme` and `uri`.

However, in practice, governments specify goods using both units of measurement and units of presentation, e.g. "5 pallets of toilet paper", "1 ton of gravel", etc.

As such, we should clarify the descriptions to explicitly allow for both types of units. We can also check that the classification scheme codelist covers units of presentation.

### PR DESCRIPTION
**Pull request by odscjen**
closes #1343 

draft until [comment](https://github.com/open-contracting/standard/issues/1343#issuecomment-2112108336) is agreed upon

### CODE REVIEW

File: docs/schema/reference.md [hunk 1]
```diff
@@ -537,11 +537,11 @@ The items block is used to list the line-items associated with a tender, award o
 
 #### Unit
 
-The `unit` block allows detailed specification of the parameters and price of units that make up a line-item.
+The `unit` block allows detailed specification of the parameters and price of units that make up a line-item. It should be used for units of measurement only. For units of presentation it is recommended to use the `ImmediateContainer` subschema.
```
Comment (jpmckinney) [inline docs/schema/reference.md]: We probably ought to define unit of measurement for clarity.

```suggestion
The `unit` subschema allows detailed specification of the parameters and price of units that make up a line-item. It should be used for units of measurement only. For units of presentation, use the [`ImmediateContainer` subschema](#immediatecontainer), which is used by the [`Item.immediateContainer` field](#item).
```

File: docs/schema/reference.md [hunk 2]
```diff
@@ -558,6 +558,25 @@ Other unit classification schemes can be used, including those in the [unitClass
 :collapse: value
 ```
 
+#### ImmediateContainer
+
+The `ImmediateContainer` subschema allows detailed specification of the immediate container of an item. This is also known as the presentation, the presentation form, the package. For units of measurement it is recommended to use the `Unit` subschema.
```
Comment (jpmckinney) [inline docs/schema/reference.md]:
```suggestion
The `ImmediateContainer` subschema allows detailed specification of the immediate container of an item. This is also called the presentation, presentation form, or package. For units of measurement, use the [`Unit` subschema](#unit).
```

**Comment by jpmckinney** [comment]
Just a couple comments on the Markdown. I'll review the schema when ready.

File: docs/schema/reference.md (hunk 1 above)
Comment (odscjen) [reply docs/schema/reference.md]: have added in a definition here of what a unit of measurement is as well as accepting this suggestion

**Comment by odscjen** [comment]
Error
FAILED tests/test_schema_integrity.py::test_versioned_release_schema_is_in_sync - UserWarning: /definitions/Quantity/properties/value has unrecognized type ['string', 'number', 'null']

Quantity has been lifted straight from the medicine extension, I think the test may need updating

**Comment by jpmckinney** [comment]
Noting that I'll correct a "agreeemnts" typo and update the profile template via this PR before merging.

**Comment by jpmckinney** [comment]
I considered adding 'GDSN' to `itemClassificationScheme.csv` (similarly, we could add a code for the immediateContainer codelist from the medicine extension), but since we discourage units of presentation for the `unit` field, I think it makes sense to omit it.

**Review by jpmckinney — APPROVED** [review]
I made edits as described in commit messages. @duncandewhurst You might want to review commits individually, skipping the chore and ci commits.

**Comment by duncandewhurst** [comment]
I reviewed the individual commits and ran the pre-commit script to update the dereferenced release schema so that I could review the HTML documentation.

The changes look good, but the first sentence of the descriptions of `Item.unit` and `Item.immediateContainer` simply restate the title of the fields. Perhaps they could be improved? e.g.

* `.unit`: "The pricing unit" -> "The unit in which the pricing of the items is specified"
* `.immediateContainer`: "The immediate container for the item" -> "The innermost container in which the items are packaged"

Anyway, happy for you to merge if you think the descriptions are OK as-is.

**Comment by jpmckinney** [comment]
Good call - I'll reword to be more informative.

**Comment by jpmckinney** [comment]
> The innermost container in which the items are packaged

For immediate container:

ISO_11239_2012(en).PDF

> immediate packaging in which a manufactured item or pharmaceutical product is contained and with which it is in direct contact
> EXAMPLES Ampoule, vial, prefilled syringe, bottle, blister.
> NOTE 1 An immediate container can be fitted with or have integrated into it an administration device and/or closure.
> NOTE 2 A pharmaceutical dose form can fulfil the role of an immediate container, e.g. a capsule containing a powder for inhalation; the capsule in this case is not a container.
> NOTE 3 An alternative, compatible definition of immediate container (“immediate packaging”) is given in Directive 92/27/EEC.
> NOTE 4 Adapted from ENV 12610:1997.

[Directive 92/27/EEC](https://eur-lex.europa.eu/legal-content/en/ALL/?uri=CELEX%3A31992L0027)

> immediate packaging means the container or other form of packaging immediately in contact with the medicinal product

So I'll go with:

> The innermost packaging with which the item is in direct contact.

**Comment by jpmckinney** [comment]
And I'll simplify to "The unit for pricing the item".

**Pull request merged**